lib/chat_models/chat_vertex_ai.ex

```diff
@@ -128,8 +123,11 @@ defmodule LangChain.ChatModels.ChatVertexAI do
   end
 
   def for_api(%ChatVertexAI{} = vertex_ai, messages, functions) do
+    sys_instructions = messages |> Enum.find(fn x -> x.role == :system end) |> List.wrap()
+    other_messages = Enum.reject(messages, fn x -> x in sys_instructions end) |> List.wrap()
```

Comment:
This can be done using `LangChain.Utils.split_system_message/1`

```elixir
{sys_instructions, other_messages} = Utils.split_system_message(messages)
```